ui/insight/src/info.ts

```diff
@@ -1,45 +1,59 @@
-var m = require('mithril');
+import { onInsert } from 'common/snabbdom';
+import { h } from 'snabbdom';
+import Ctrl from './ctrl';
 
-var shareStates = ['nobody', 'friends only', 'everybody'];
+const shareStates = ['nobody', 'friends only', 'everybody'];
 
-module.exports = function (ctrl) {
-  var shareText = 'Shared with ' + shareStates[ctrl.user.shareId] + '.';
-  return m('div.info.box', [
-    m('div.top', [
-      m(
-        'a.username.user-link.insight-ulpt',
-        {
-          href: '/@/' + ctrl.user.name,
-        },
-        ctrl.user.name
-      ),
-    ]),
-    m('div.content', [
-      m('p', ['Insights over ', m('strong', ctrl.user.nbGames), ' rated games.']),
-      m(
+export default function (ctrl: Ctrl) {
+  const shareText = 'Shared with ' + shareStates[ctrl.user.shareId] + '.';
+  return h('div.info.box', [
+    h('div.top', [h('a.username.user-link.insight-ulpt', { attrs: { href: '/@/' + ctrl.user.name } }, ctrl.user.name)]),
+    h('div.content', [
+      h('p', ['Insights over ', h('strong', ctrl.user.nbGames), ' rated games.']),
+      h(
         'p.share',
         ctrl.own
-          ? m(
+          ? h(
               'a',
               {
-                href: '/account/preferences/privacy',
-                target: '_blank',
-                rel: 'noopener',
+                attrs: {
+                  href: '/account/preferences/privacy',
+                  target: '_blank',
+                  rel: 'noopener',
+                },
               },
               shareText
             )
           : shareText
       ),
     ]),
-    m('div.refresh', {
-      config: function (e, isUpdate) {
-        if (isUpdate) return;
-        var $ref = $('.insight-stale');
-        if ($ref.length) {
-          $(e).html($ref.show());
-          lichess.refreshInsightForm();
-        }
-      },
-    }),
+    h(
+      'div.refresh',
+      ctrl.env.user.stale
+        ? h('div.insight-stale', [
+            h('p', 'There are new games to learn from!'),
+            h(
+              'form.insight-refresh',
+              {
+                attrs: {
+                  action: `/insights/refresh/${ctrl.env.user.id}`,
+                  method: 'post',
+                },
+                hook: onInsert(_el => lichess.refreshInsightForm()),
+              },
+              [
+                h('button.button.text', { attrs: { 'data-icon': '' } }, 'Upate insights'),
+                h(
+                  'div.crunching.none',
+                  {
+                    hook: onInsert(el => el.insertAdjacentHTML('afterbegin', lichess.spinnerHtml)),
+                  },
+                  [h('br'), h('p', h('strong', 'Now crunching data just for you!'))]
+                ),
+              ]
+            ),
+          ])
+        : null
+    ),
```

Comment:
The original code implements the refresh form by grabbing the element from the html coming from the server persisting it in a way that bypassed the vdom. I wasn't able to replicate this in snabbdom, so the form is now part of the vdom.